### PR TITLE
placement: refine the log errs in placement

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

To solve #2704 

### Check List

Tests

- Unit test
- Integration test

### Release note

- No release note

### CODE REVIEW

**Comment by JmPotato** [comment]
@ZenoTan PTAL

File: server/schedule/placement/rule_list.go [hunk 1]
```diff
@@ -114,9 +115,10 @@ func buildRuleList(rules map[[2]string]*Rule) (ruleList, error) {
 				if i != len(points)-1 {
 					endKey = points[i+1].key
 				}
-				return ruleList{}, errors.Errorf("no rule for range {%s, %s}",
+				return ruleList{}, errs.ErrBuildRuleList.FastGenByArgs(fmt.Sprintf("no rule for range {%s, %s}",
 					strings.ToUpper(hex.EncodeToString(p.key)),
-					strings.ToUpper(hex.EncodeToString(endKey)))
+					strings.ToUpper(hex.EncodeToString(endKey))))
+
```
Comment (rleungx) [inline server/schedule/placement/rule_list.go:121]: remove the empty line

**Comment by nolouch** [comment]
LGTM

**Pull request merged**